tools/preprocessor.js

```diff
@@ -55,6 +55,12 @@ const inputFile = arguments_[1];
 const expandMacros = arguments_.includes('--expandMacros');
 
 load(settingsFile);
+
+if (!ALL_INCOMING_MODULE_JS_API.length) {
+  ALL_INCOMING_MODULE_JS_API = INCOMING_MODULE_JS_API;
+}
+INCOMING_MODULE_JS_API = new Set(INCOMING_MODULE_JS_API);
```

Comment:
Why is this part needed?

Reply:
Because of this line: https://github.com/emscripten-core/emscripten/blob/fcb33368d64873786762df2dd44cc9a18d4b1286/src/parseTools.js#L752

It expects a `Set`. `compiler.js` does the same.

Reply:
It looks like `compiler.js` turns both `INCOMING_MODULE_JS_API` and `ALL_INCOMING_MODULE_JS_API` into sets:

https://github.com/emscripten-core/emscripten/blob/a7d1f86b121e4223787db99ff9134ad964b08ddf/src/compiler.js#L76-L77

But this code only does that for the former. Why is there a difference?

Reply:
Since I used only `INCOMING_MODULE_JS_API` I did not add the rest.

Reply:
But I'm curious why this change required this bug fix?  Does it fail without this change?

Reply:
Yes, an array does not have `.has()`, only `Set` does